src/posts/components/PostsList.tsx

```diff
@@ -1,76 +1,94 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { DataTable } from "primereact/datatable";
 import { Column } from "primereact/column";
 import dayjs from "dayjs";
-import { posts } from "../posts-data";
-import { Post } from "../posts.types";
 import clsx from "clsx";
 import "../../assets/styles/responsive.scss";
-
 import "primereact/resources/primereact.min.css";
 import "primereact/resources/themes/saga-blue/theme.css";
 import "bootstrap/dist/css/bootstrap.min.css";
+import { Post } from "../posts.types";
+import { getAllPosts, removePost } from "../posts-api";
 
 const PostsList: React.FC = () => {
-  const [data] = useState<Post[]>(posts);
+  const [posts, setPosts] = useState<Post[]>([]);
 
-  //render status badge
-  const statusTemplate = (rowData: Post) => {
-    return (
-      <span
-        className={clsx(
-          "badge p-2 text-uppercase",
-          rowData.status === "published" && "bg-success",
-          rowData.status === "draft" && "bg-warning text-dark",
-          rowData.status === "deleted" && "bg-danger"
-        )}
-      >
-        {rowData.status.toUpperCase()}
-      </span>
-    );
+  //fetch posts on component mount
+  useEffect(() => {
+    const fetchPosts = async () => {
+      const allPosts = await getAllPosts();
+      setPosts(allPosts);
+    };
+    fetchPosts();
+  }, []);
+
+  //handle post deletion
+  const handleDelete = async (id: number) => {
+    await removePost(id);
```

Comment:
add error handler using try catch, if there is an error, show it on toast message (https://www.npmjs.com/package/react-toastify),

add error handling to all used api methods